plugin/core/types.py

```diff
@@ -621,6 +678,27 @@ def match_view(self, view: sublime.View) -> bool:
             return sublime.score_selector(syntax.scope, self.selector) >= 8
         return False
 
+    def map_client_path_to_server_uri(self, path: str) -> str:
+        if self.path_maps:
+            for path_map in self.path_maps:
+                path, mapped = path_map.map_from_local_to_remote(path)
+                if mapped:
+                    break
+        return urllib.parse.urljoin('file:', urllib.request.pathname2url(path))
+
+    def map_server_uri_to_client_path(self, uri: str) -> str:
+        if os.name == 'nt':
+            # url2pathname does not understand %3A (VS Code's encoding forced on all servers :/)
+            path = urllib.request.url2pathname(urllib.parse.urlparse(uri).path).strip('\\')
+        else:
+            path = urllib.request.url2pathname(urllib.parse.urlparse(uri).path)
```

Comment:
I wonder what cases it's trying to handle. Some tests would be nice to have to understand it better. But I guess Windows tests would have to run on Windows?

Reply:
I took this directly from our good ol' `uri_to_filename` function in plugin/core/url.py

Reply:
Why are we not using it directly then? :)

Reply:
This is meant to replace `uri_to_filename` and `filename_to_uri` eventually. But for the time being I'll reuse the functions.